### PR TITLE
migration DDL preview functionality

### PR DESCRIPTION
**Pull request by AliSayyah**
Hey there!
Just some cleanups and initial base stuff.
Doesn't work right now.

### CODE REVIEW

**Comment by dantownsend** [comment]
@AliSayyah This looks great, thanks!

It's exactly what I had in mind.

**Comment by AliSayyah** [comment]
I'll work on more tests, but how do you think we should handle the command?
```

**Comment by dantownsend** [comment]
> I'll work on more tests, but how do you think we should handle the command?
> 
> piccolo migrations forwards my_app --preview
> piccolo migrations backwards 2018-09-04T19:44:09 --preview
> Would this work?

Yes, I like that 👍

**Comment by AliSayyah** [comment]
Does `targ` support command options like `--preview`?

**Comment by AliSayyah** [comment]
Hey, @dantownsend. This seems ok to me. But let me know if there are any problems, or the tests are insufficient.

File: piccolo/apps/migrations/auto/migration_manager.py [hunk 1]
```diff
@@ -739,16 +770,17 @@ async def run_backwards(self):
...
-                else:
-                    raw()
-
-            await self._run_add_columns(backwards=True)
-            await self._run_add_tables(backwards=True)
-            await self._run_drop_tables(backwards=True)
-            await self._run_rename_tables(backwards=True)
-            await self._run_drop_columns(backwards=True)
-            await self._run_rename_columns(backwards=True)
-            await self._run_alter_columns(backwards=True)
```
Comment (dantownsend) [inline piccolo/apps/migrations/auto/migration_manager.py:754]: It looks good - my only concern was this bit.

You've merged `run` and `run_backwards` - I can understand why, as it reduces repetition. The order of the operations were slightly different before when running backwards. Having thought about it though, I think the new order should also be fine.

**Comment by dantownsend** [comment]
@AliSayyah This looks good to me, thanks.

I think we should release it.

**Pull request merged**